lib/puppet/resource/type.rb

```diff
@@ -453,7 +455,9 @@ def convert_from_ast(name)
   end
 
   def evaluate_parent_type(resource)
-    return unless klass = parent_type(resource.scope) and parent_resource = resource.scope.compiler.catalog.resource(:class, klass.name) || resource.scope.compiler.catalog.resource(:node, klass.name)
+    klass = parent_type(resource.scope)
+    parent_resource = resource.scope.compiler.catalog.resource(:class, klass.name) || resource.scope.compiler.catalog.resource(:node, klass.name) if klass
```

Comment:
We need to check for nil `klass` before calling `klass.name` on line 459.

Reply:
there's a postfix if at the end of this line